test/correctness/simd_op_check.h

```diff
@@ -208,7 +210,7 @@ class SimdOpCheckTest {
             g.compute_at(f, x)
                 .update()
                 .split(x, xo, xi, vector_width)
-                .atomic()
+                .atomic(has_inline_reduction.override_associativity_test)
```

Comment:
Thinking about this more, this does indicate that saturating dot products are kind of a problem... are they actually useful? It seems like only an unsigned saturating sum would be useful, a signed saturating sum is going to do weird things when there are mixes of positive and negative terms in the summation. (But this instruction set has only signed saturating sums!)

The result could saturate at the max and add a bunch of positive terms, and then a bunch of negative terms, then the result will be something far from the max, but that's incorrect if the positive terms are bigger than the negative terms.

Reply:
My understanding was that it's only unsigned saturation dot products that are really useful. signed ones aren't even associative!

Reply:
From my understanding of the operation, at the instruction level, the saturation part occurs at the end, after all the results are multiplied to each other, so the associativity of the summation terms is not really a problem. A signing saturating sum is indeed not associative, but with the ability to override the associativity test makes it explicit that you know what you are doing.

I didn't want to set it to always be true to avoid interfering with existing tests.

> My understanding was that it's only unsigned saturating dot products that are really useful. signed ones aren't even associative!

I don't know much about the field but saturating dot products, both signed and unsigned, are used in VoIP applications, but you are right, they are not associative

Reply:
atomic(true) won't break anything that atomic() didn't. Basically setting that to always true reduces coverage of atomic()'s associativity test, but IMO it's better to just accept that (very slight) loss in coverage to avoid the extra logic/possibly brittle test behavior.

Regarding associativity: I agree if this instruction were only a 4-way dot product, the saturation is uninteresting. But this is an accumulating dot product instruction, so it can be used in an arbitrarily large loop, and presumably it would be in ML applications (I assume the reason this instruction is being added). In that case, the saturation is hard to reason about.

Reply:
> atomic(true) won't break anything that atomic() didn't. Basically setting that to always true reduces coverage of atomic()'s associativity test, but IMO it's better to just accept that (very slight) loss in coverage to avoid the extra logic/possibly brittle test behavior.

That sounds reasonable, I've set it to true in a1d2685499d1779bde36a8323c61235a8f232c81